app/(gcforms)/[locale]/(form administration)/form-builder/[id]/responses/[statusFilter]/components/ManageFormAccessDialog/ManageUsers.tsx

```diff
@@ -0,0 +1,262 @@
+        </label>
+        <p className="pb-3">{t("mustBeAGovernmentAddress")}</p>
+
+        {errors.length > 0 && (
+          <div className="my-2 flex flex-wrap gap-2">
+            {errors.map((error, index) => (
+              <div
+                className="rounded-md border border-red-700 bg-red-100 px-2 text-red-700"
+                key={`error-${index}`}
+              >
+                {error}
+              </div>
+            ))}
+          </div>
+        )}
+
+        <div className="flex flex-wrap gap-2 border border-black p-2">
+          {emailList.map((email) => {
+            return (
+              <div
+                key={email}
+                className="flex items-center gap-1 rounded-md border border-violet-700 bg-violet-50 px-2"
+              >
+                <div className="">{email}</div>
+                <button className="" onClick={() => handleRemoveEmail(email)}>
+                  <CancelIcon className="size-5" />
+                </button>
+              </div>
+            );
+          })}
+
+          <input
+            id="email"
+            type="text"
+            className="inline-block grow border-none px-2 py-1 outline-none"
+            onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+              setSelectedEmail(e.target.value);
+            }}
+            onKeyDown={(e: React.KeyboardEvent<HTMLInputElement>) => {
+              if (e.key === "Backspace" && e.currentTarget.value === "") {
+                handleRemoveEmail(emailList[emailList.length - 1]);
+              }
+              if (e.key === "," || e.key === " " || e.key === ";" || e.key === "Enter") {
+                e.preventDefault();
+                handleAddEmail(e.currentTarget.value);
+              }
+              if (e.key === "Tab" && e.currentTarget.value !== "") {
+                e.preventDefault();
+                handleAddEmail(e.currentTarget.value);
+              }
+            }}
+            value={selectedEmail}
+          />
+        </div>
+      </section>
+
+      <section className="mt-4">
+        <h3>{t("peopleWithAccess")}</h3>
+        <div className="max-h-96 overflow-y-scroll border-1 border-black">
+          {loading && <div className="p-10">{t("loading")}</div>}
+          {!loading && (
+            <>
+              {usersWithAccess.map((user, index) => (
+                <div
+                  className={cn(
+                    "flex flex-row items-start px-4 py-2",
+                    (loggedInUserEmail === user.email ||
+                      (index === 0 &&
+                        usersWithAccess.filter((u) => !hasOwnProperty(u, "expired")).length <=
+                          1)) &&
+                      "bg-slate-100 py-4"
+                  )}
+                  key={user.email}
+                >
+                  <div className="grow">{user.email}</div>
+                  {hasOwnProperty(user, "expired") ? (
+                    <div>
```

Comment:
Might benefit here by breaking down some components unless passing props gets in the way.

Reply:
https://github.com/cds-snc/platform-forms-client/pull/4392/commits/b4ef546d6c4a266a287331fa809ba32f6e71efdb and https://github.com/cds-snc/platform-forms-client/pull/4392/commits/98fc923361292b32c3408c710f179d29073785d1